var/spack/repos/builtin/packages/rocblas/package.py

```diff
@@ -29,27 +29,28 @@ class Rocblas(CMakePackage):
     version('3.7.0', sha256='9425db5f8e8b6f7fb172d09e2a360025b63a4e54414607709efc5acb28819642', deprecated=True)
     version('3.5.0', sha256='8560fabef7f13e8d67da997de2295399f6ec595edfd77e452978c140d5f936f0', deprecated=True)
 
-    tensile_architecture = ('all', 'gfx906', 'gfx908', 'gfx803', 'gfx900',
-                            'gfx906:xnack-', 'gfx908:xnack-', 'gfx90a:xnack+',
-                            'gfx90a:xnack-', 'gfx1010', 'gfx1011',
-                            'gfx1012', 'gfx1030')
+    amdgpu_targets = ('gfx906', 'gfx908', 'gfx803', 'gfx900',
+                      'gfx906:xnack-', 'gfx908:xnack-', 'gfx90a:xnack+',
+                      'gfx90a:xnack-', 'gfx1010', 'gfx1011',
+                      'gfx1012', 'gfx1030')
 
-    variant('tensile_architecture', default='all', values=tensile_architecture, multi=True)
+    variant('amdgpu_target', default='gfx906:xnack-', values=amdgpu_targets, multi=True)
```

Comment:
I really don't like defaulting to one random architecture. Yeah, I have gfx906 in my workstation, but now everyone on W6800, MI100, MI200, etc. needs to manually specify the architecture. And they probably will only find out they need to do that by building everything, running their program, having it fail, and debugging what's missing.

Reply:
@teonnik  Tensile is already in a separate package called 
rocm-tensile.

Reply:
@srekolam Thanks! I missed that one! I will try to rewrite the PR in terms of that package.

Reply:
I'm not sure that makes sense. Tensile exists in three forms:
1. It's a command-line executable for generating kernels, benchmarking them saving the best solutions in YAML files.
2. It's a build system component that reads YAML solution files, generates the specified kernels, and invokes the compiler to build them. It stores the results of that process in TensileLibrary.dat, and various code object files (`*.co`, `*.hsaco`).
3. It's a run-time library that will load the TensileLibrary.dat and code object files when you call it do do a GEMM.

I could be totally wrong, but I would think rocm-tensile only fulfills role (1). In any case, the YAML solution files are a part of rocBLAS, not Tensile (i.e., they are solutions to rocBLAS's problems). You won't save any time building Tensile separately, because (2) is what takes so long, and it needs rocBLAS's data. So, there's not much benefit to separating them.

I'm also not sure if (3) can be built ahead of time. @lringham, can you use the built version of rocm-tensile to build rocBLAS, or does it _require_ the Tensile source? For reference, [this is how Tensile is being built in the separate package](https://github.com/spack/spack/blob/66c70b82927fc7d3e37009461407fe282873a9a9/var/spack/repos/builtin/packages/rocm-tensile/package.py).

Reply:
I forwarded this on to one of the senior members of the rocBLAS team, who responded "if they build rocBLAS and Tensile differently than us, [...] it will be a maintenance problem to support them."

I would not recommend using a separate rocm-tensile.

Reply:
@cgmb OK, I will not try to use `rocm-tensile` as a dependency in this package. I will also try to return the previous previous behavior which was to build all architectures by default. I would like to do that without an explicit `all` value but I am not sure if that is possible at the moment. I will have a go at it next week.

Reply:
I opened #28907 for the `AMDGPU_TARGETS` defaults and propagation change.

Reply:
@cgmb  I see that you have incorporated most of the changes from this PR. Perhaps it is better to continue on your PR and close this one?